python/lsst/daf/butler/registry/queries/_query.py

```diff
@@ -613,3 +682,112 @@ def makeBuilder(self, summary: Optional[QuerySummary] = None) -> QueryBuilder:
         builder.joinTable(self.sql.alias(), dimensions=self.graph.dimensions,
                           datasets=self.getDatasetColumns())
         return builder
+
+
+class MaterializedQuery(Query):
+    """A `Query` implementation that represents query results saved in a
+    temporary table.
+
+    `MaterializedQuery` instances should not be constructed directly; use
+    `Query.materialize()` instead.
+
+    Parameters
+    ----------
+    table : `sqlalchemy.schema.Table`
+        SQLAlchemy object represnting the temporary table.
+    spatial : `Iterable` [ `DimensionElement` ]
+        Spatial dimension elements whose regions must overlap for each valid
+        result row (which may reject some rows that are in the table).
+    datasetType : `DatasetType`
+        The `DatasetType` of datasets returned by this query, or `None`
+        if there are no dataset results
+    isUnique : `bool`
+        If `True`, the table's rows are unique, and there is no need to
+        add ``SELECT DISTINCT`` to gaurantee this in results.
+    graph : `DimensionGraph`
+        Dimensions included in the columns of this table.
+    whereRegion : `Region` or `None`
+        A spatial region all result-row regions must overlap to be valid (which
+        may reject some rows that are in the table).
+    managers : `RegistryManagers`
+        A struct containing `Registry` manager helper objects, forwarded to
+        the `Query` constructor.
+    """
+    def __init__(self, *,
+                 table: sqlalchemy.schema.Table,
+                 spatial: Iterable[DimensionElement],
+                 datasetType: Optional[DatasetType],
+                 isUnique: bool,
+                 graph: DimensionGraph,
+                 whereRegion: Optional[Region],
+                 managers: RegistryManagers):
+        super().__init__(graph=graph, whereRegion=whereRegion, managers=managers)
+        self._table = table
+        self._spatial = tuple(spatial)
+        self._datasetType = datasetType
+        self._isUnique = isUnique
+
+    def isUnique(self) -> bool:
```

Comment:
It's a naming convention thing.  Because we almost always name methods starting with verbs, I don't like to start non-methods with verbs unless it's completely ambiguous that they aren't methods.  I don't know about everyone else, but this helps me remember whether or not I should add a `()` to any particular interface.

Reply:
in that case, does it need to be called `isUnique` or would `unique` as an attribute (or property) do the same thing? Does `is` add anything?

Reply:
I think it's a bit clearer.  Just "unique" in only a slightly context could be an attribute that is a container of things that are unique, rather than a bool, while "isUnique" is very clearly a bool, and I think a set of parens is a reasonable price to pay for that.

Reply:
sounds reasonable